build.c

```diff
@@ -445,8 +483,12 @@ jobdone(struct job *j)
 		j->failed = true;
 	}
 	close(j->fd);
-	if (j->buf.len && (!consoleused || j->failed))
+	if (j->buf.len && (!consoleused || j->failed)) {
+		if (!dumb)
+			putchar('\n');
 		fwrite(j->buf.data, 1, j->buf.len, stdout);
+		newline = true;
```

Comment:
Might be good to use `newline = j->buf.data[j->buf.len - 1] == '\n'` to see if the output actually ended in a newline.